### PR TITLE
add GitHub templates for issue reporting and PR

### PR DESCRIPTION
**Pull request by tennix**
GitHub allows a repository to [create issue templates and PR template under the `.github` directory](https://help.github.com/en/articles/about-issue-and-pull-request-templates). This PR adds some templates for issue reporting and PR. It should help users, contributors and ourselves to communicate more efficiently.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+<!-- You can run `kubectl exec -n tidb-admin {tidb-controller-manager-pod-name} -- tidb-controller-manager -V` -->
+
+**What storage classes are used for PD and TiKV?**
+<!-- You can run `kubectl get pvc -n {tidb-cluster-namespace}` -->
+
+**What's the status of the TiDB cluster pods?**
+<!-- You can run `kubectl get po -n {tidb-cluster-namespace} -o wide` -->
+
+**What did you do?**
+<!-- If possible, provide a recipe for reproducing the error. A complete runnable program is good. -->
```
Comment (gregwebs) [inline .github/ISSUE_TEMPLATE/bug-report.md]: I don't think we are likely to get runnable program bug reports for tidb-operator.

File: .github/ISSUE_TEMPLATE/question.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+name: "\U0001F914 Question"
+about: Usage question that isn't answered in docs or discussion
+
+---
+
+## Question
+
+Before asking a question, make sure you have:
+
+- Searched existing Stack Overflow questions.
```
Comment (gregwebs) [inline .github/ISSUE_TEMPLATE/question.md]: How is this done? What are the tags?

File: .github/ISSUE_TEMPLATE/question.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- Googled your question.
```
Comment (gregwebs) [inline .github/ISSUE_TEMPLATE/question.md]: what keywords should be used?

File: .github/ISSUE_TEMPLATE/question.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+- Searched open and closed [GitHub issues](https://github.com/pingcap/tidb-operator/issues?utf8=%E2%9C%93&q=is%3Aissue)
+- Read the documentation:
+  - [TiDB Operator Readme](https://github.com/pingcap/tidb-operator)
+  - [TiDB Operator Doc](https://github.com/pingcap/tidb-operator/tree/master/docs)
```
Comment (gregwebs) [inline .github/ISSUE_TEMPLATE/question.md]: This is being moved soon?

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+Code changes
+
+ - Has Helm charts change
+ - Has Go code change
+ - Has CI related scripts change
+
+Side effects
+
+ - Increased code complexity
```
Comment (gregwebs) [inline .github/pull_request_template.md]: How is this information used?

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 1 above)
Comment (tennix) [reply .github/ISSUE_TEMPLATE/bug-report.md]: This is meant to let the user write how he deploys tidb-operator and tidb-cluster.

File: .github/ISSUE_TEMPLATE/question.md (hunk 3 above)
Comment (tennix) [reply .github/ISSUE_TEMPLATE/question.md]: Yes, I admit that we don't have many users right now. So the search via Google and StackOverflow may not suitable right now. I can remove it if you think it's unnecessary and add it back when needed.

File: .github/ISSUE_TEMPLATE/question.md (hunk 4 above)
Comment (tennix) [reply .github/ISSUE_TEMPLATE/question.md]: Yes, we'll move our documents to https://github.com/pingcap/docs and https://github.com/pingcap/docs-cn to make our documents available on PingCAP official website.

File: .github/pull_request_template.md (hunk 5 above)
Comment (tennix) [reply .github/pull_request_template.md]: Hmm, this is copied from PD and TiDB repository. I think we share the same idea with these repos.

Comment (gregwebs) [reply .github/pull_request_template.md]: I hear complaints about the existing templates. I think we can improve upon them by deleting things of dubious value and try to have a checklist of things that submitters must do or explain. Code complexity is very subjective and will need to be assessed independently by the reviewers ignoring of what is written here. If we really want this information in a useful objective form we could run a cyclomatic complexity linter.

File: .github/ISSUE_TEMPLATE/question.md (hunk 4 above)
Comment (gregwebs) [reply .github/ISSUE_TEMPLATE/question.md]: Then this link will break?

File: .github/ISSUE_TEMPLATE/question.md (hunk 3 above)
Comment (gregwebs) [reply .github/ISSUE_TEMPLATE/question.md]:
```suggestion
- Reviewed relevant Kubernetes information: Google your error messages and look at K8s docs.
```

Comment (gregwebs) [reply .github/ISSUE_TEMPLATE/question.md]: Yes, we should decide if we want to use Stack Overflow or github issues.

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 1 above)
Comment (gregwebs) [reply .github/ISSUE_TEMPLATE/bug-report.md]:
```suggestion
<!-- If possible, provide a recipe for reproducing the error. -->
```

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+about: Something isn't working as expected
+
+---
+
+## Bug Report
+
+**What version of Kubernetes are you using?**
+<!-- You can run `kubectl version` -->
+
+**What storage classes exist in the Kubernetes cluster?**
```
Comment (gregwebs) [inline .github/ISSUE_TEMPLATE/bug-report.md]: Group/Combine this with the below storage class question about PD/TiKV.

File: .github/ISSUE_TEMPLATE/question.md (hunk 4 above)
Comment (tennix) [reply .github/ISSUE_TEMPLATE/question.md]: We should update the links when our documents are all moved to the pingcap/docs and pingcap/docs-cn.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+### What problem does this PR solve? <!--add issue link with summary if exists-->
+
+### What is changed and how it works?
+
+### Check List <!--REMOVE the items that are not applicable-->
+
+Tests <!-- At least one of them must be included. -->
+
+ - Unit test
+ - E2E test
```
Comment (weekface) [inline .github/pull_request_template.md]: add stability test?

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
+<!--
+Thank you for contributing to TiDB Operator! Please read TiDB Operator's [CONTRIBUTING](https://github.com/pingcap/tidb-operator/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
+-->
+
+### What problem does this PR solve? <!--add issue link with summary if exists-->
```
Comment (lilin90) [inline .github/pull_request_template.md]:
```suggestion
### What problem does this PR solve? <!--add an issue link with a summary if it exists-->
```

**Pull request merged**